packages/kit/types/ambient.d.ts

```diff
@@ -113,10 +113,22 @@ declare module '$app/navigation' {
 		opts?: { replaceState?: boolean; noscroll?: boolean; keepfocus?: boolean; state?: any }
 	): Promise<void>;
 	/**
-	 * Causes any `load` functions belonging to the currently active page to re-run if they `fetch` the resource in question. If no argument is given, all resources will be invalidated. Returns a `Promise` that resolves when the page is subsequently updated.
+	 * Causes any `load` functions belonging to the currently active page to re-run if they registered it with `fetch` or `depends`. Returns a `Promise` that resolves when the page is subsequently updated.
+	 *
+	 * If no argument is given, all resources will be invalidated.
+	 *
+	 * The `string` argument has to match the URI that was passed to `fetch` or `depends` exactly (including query parameters).
+	 *
+	 * The `function` argument can be used define a custom predicate. It receives the full URI and causes `load` to rerun if `true` is returned.
+	 * This can be useful if you want to invalidate based on a pattern instead of a exact match.
+	 *
+	 * ```js
+	 * // Example: Match '/path' regardless of the query parameters
+	 * invalidate((url) => (new URL(url)).pathname === '/path')`
```

Comment:
This would be way nicer if `url` was already a `URL` object rather than a string. Might change that alongside adding `invalidateAll()`

Reply:
True. `invalidate((url) => url.pathname === '/path')` or `invalidate(({pathname}) => pathname === '/path')` is a lot more readable and getting full url as string would still be possible with `url.href`.

Sadly I dont have the time to implement this my self at the moment.

Reply:
implemented in #6493